corehq/apps/cloudcare/views.py

```diff
@@ -98,11 +101,52 @@ def get(self, request, domain):
         )
         apps = filter(None, apps)
         apps = filter(lambda app: app.get('cloudcare_enabled') or self.preview, apps)
-        apps = filter(lambda app: app_access.user_can_access_app(request.couch_user, app), apps)
-        role = request.couch_user.get_role(domain)
+        apps = filter(lambda app: app_access.user_can_access_app(user, app), apps)
+        role = user.get_role(domain)
         if role:
             apps = [app for app in apps if role.permissions.view_web_app(app)]
         apps = sorted(apps, key=lambda app: app['name'])
+        return apps
+
+    @staticmethod
+    def get_restore_as_user(request, domain):
+        """
+        returns (user, set_cookie), where set_cookie is a function to be called on
+        the eventual response
+        """
+
+        def set_cookie(response):  # set_coookie is a noop by default
+            return response
+
+        cookie_name = urllib.quote(
+            'restoreAs:{}:{}'.format(domain, request.couch_user.username))
+        username = request.COOKIES.get(cookie_name)
+        if username:
+            user = CouchUser.get_by_username(format_username(username, domain))
+            if user:
+                return user, set_cookie
+            else:
+                def set_cookie(response):  # overwrite the default noop set_cookie
```

Comment:
this kind of confused me when i first read it. would it make more sense to just return a boolean that says whether or not to delete the cookie?